chapters/sampling-randomization-power.tex

```diff
@@ -1,427 +1,759 @@
 via the master do-file may produce different results,
 since Stata's \texttt{version} setting expires after each time you run your do-files.
 
+% rule 2: sorting
 \textbf{Sorting} means that the actual data that the random process is run on is fixed.
-Because random numbers are assigned to each observation in row-by-row starting from
+Because random numbers are assigned to each observation row-by-row starting from
 the top row,
 changing their order will change the result of the process.
-Since the exact order must be unchanged, the underlying data itself must be unchanged as well between runs.
-This means that if you expect the number of observations to change (for example increase during
-ongoing data collection) your randomization will not be stable unless you split your data up into
-smaller fixed datasets where the number of observations does not change. You can combine all
-those smaller datasets after your randomization.
 In Stata, the only way to guarantee a unique sorting order is to use
-\texttt{isid [id\_variable], sort}. (The \texttt{sort, stable} command is insufficient.)
-You can additionally use the \texttt{datasignature} command to make sure the
-data is unchanged.
+\texttt{isid [id\_variable], sort}.
+(The \texttt{sort, stable} command is insufficient.)
+Since the exact order must be unchanged,
+the underlying data itself must be unchanged as well between runs.
+This means that if you expect the number of observations to change
+(for example to increase during ongoing data collection),
+your randomization will not be stable unless you split your data up into
+smaller fixed datasets where the number of observations does not change.
+You can combine all
+those smaller datasets after your randomization.
+
 
-\textbf{Seeding} means manually setting the start-point in the list of random numbers.
-The seed is a number that should be at least six digits long and you should use exactly
+% rule 3: seeding
+\textbf{Seeding} means manually setting the start point in the list of random numbers.
+A seed is just a single number that specifies one of the possible start points.
+It should be at least six digits long and you should use exactly
 one unique, different, and randomly created seed per randomization process.\sidenote{You
-can draw a uniformly distributed six-digit seed randomly by visiting \url{https://bit.ly/stata-random}.
-(This link is a just shortcut to request such a random seed on \url{https://www.random.org}.)
-There are many more seeds possible but this is a large enough set for most purposes.}
-In Stata, \texttt{set seed [seed]} will set the generator to that start-point. In R, the \texttt{set.seed} function does the same.
+	can draw a uniformly distributed six-digit seed randomly by visiting \url{https://bit.ly/stata-random}.
+	(This link is a just shortcut to request such a random number on \url{https://www.random.org}.)
+	There are many more seeds possible but this is a large enough set for most purposes.}
```

Comment:
@bbdaniels suggestion: long side note, should we put int into the text?

Reply:
suggest rephrasing to: You can draw a uniformly distributed six-digit seed randomly from random.org (\url{https://bit.ly/stata-random}). Six digits is typically sufficient.

Reply:
why 6? is there any rule of thumb we can give for why that's sufficient?

Reply:
6 digits is commonly used as security codes when the code should rarely be repeated but should be possible to handle for a human brain. For example, if a website texts you a security code it is often 6 digits long (or the WB hard token). And you can memorize the code for ten seconds by repeating in your brain "twenty-four sixty-seven twelve" while you transfer the digits somewhere. 8 digits is so much more prone to memorizing errors.

Not that you need to memorize seeds often, but 6 digits is just a good balance of unique and easy-of-use. But I do not have a better rule of thumb than that.

Reply:
ok cool. i don't have a different rule of thumb, so leaving it at 6 digits seems fine to me.  stata's recommendation is an element chosen randomly from the set {0, 1, . . . , 2,147,483,647}.